forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt

```diff
@@ -127,7 +153,49 @@ class ForageTerminalSDK(
     override suspend fun checkBalance(
         params: CheckBalanceParams
     ): ForageApiResponse<String> {
-        return ForageApiResponse.Success("TODO")
+        val logger = createLogger()
+        val forageSdkInstance = forageSdk as ForageSDK
+        val (foragePinEditText, paymentMethodRef) = params
+        val (merchantId, sessionToken) = forageSdkInstance._getForageConfigOrThrow(foragePinEditText)
+
+        // TODO: replace Log.getInstance() with Log() in future PR
+        logger.i(
+            "[POS] Called checkBalance for PaymentMethod $paymentMethodRef",
+            attributes = mapOf(
+                "merchant_ref" to merchantId,
+                "payment_method_ref" to paymentMethodRef
+            )
+        )
+
+        // This block is used for tracking Metrics!
+        // ------------------------------------------------------
+        val measurement = CustomerPerceivedResponseMonitor.newMeasurement(
+            vault = foragePinEditText.getCollector(merchantId).getVaultType(),
+            vaultAction = UserAction.BALANCE,
+            logger
+        )
+        measurement.start()
+        // ------------------------------------------------------
+
+        val serviceFactory = createServiceFactory(sessionToken, merchantId, logger)
+        val balanceCheckService = serviceFactory.createCheckBalanceRepository(foragePinEditText)
+        val balanceResponse = balanceCheckService.posCheckBalance(
+            paymentMethodRef = paymentMethodRef,
+            posTerminalId = posTerminalId
+        )
+        forageSdkInstance.processApiResponseForMetrics(balanceResponse, measurement)
+
+        if (balanceResponse is ForageApiResponse.Failure) {
+            logger.e(
+                "[POS] checkBalance failed for PaymentMethod $paymentMethodRef on Terminal $posTerminalId ${balanceResponse.errors[0]}",
```

Comment:
Nit: Can you add a colon or something after the `$posTerminalId` to separate the error?

Reply:
ff835b192bb21e08bfed956cc3cf8b8801b70dba